### PR TITLE
PR: Update style and contributing guide to reflect Spyder 4 release, recent changes to the repo and be more usable for new contributors

### LINKED ISSUE #139: Update style guide
## Issue Description

 We decided to make some changes regarding the spacing and format, to match better the new theme so we need to fix the style guide to match these changes.

### Other information

* Is the problem specific to documentation for a particular version of Spyder?

* Are you planning on submitting a Github pull request with a suitable change?

* Anything else you'd like to add?

### PR DESCRIPTION
**Pull request by CAM-Gerlach**
### Pull Request Checklist

* [x] Read and followed this repo's [Contributing Guidelines](https://github.com/spyder-ide/spyder/blob/master/CONTRIBUTING.md)
* [x] Based your PR on the latest version of the correct branch (master or 3.x)
* [x] Checked your writing carefully for correct English spelling, grammar, etc
* [x] Described your changes and the motivation for them below
* [x] Noted what issue(s) this pull request resolves, creating one if needed

## Description of Changes

I've made a number of updates to the contributing and style guides to make them up to date with the Spyder 4 release, with changes to the repo over the past year+, and fix various minor issues. I've also broken the Contributing guide by sentence, added a section summarizing the most important points from the style guide leaving the latter more as a reference, and added a prominent link in both documents to the Sphinx reST primer.

I changed the line breaks with the first commit, so I'd strongly advise disincluding it when reviewing to make seeing what I actually changed much easier.

### Issue(s) Resolved

Fixes #139

### CODE REVIEW

**Comment by ccordoba12** [comment]
I think @juanis2112 should review this one and give her approval.

**Comment by CAM-Gerlach** [comment]
> I think @juanis2112 should review this one and give her approval.

image

**Comment by ccordoba12** [comment]
I check if she was assigned before posting (probably the Github mobile version doesn't report that).

File: STYLEGUIDE.md [hunk 1]
```diff
@@ -71,10 +72,11 @@ Must (not), should (not), recommended, suggested, may, and can generally follow
 ## Images/Figures
 
 - Lower case file names and extensions, separate with hyphens, 3-character extensions
-- For ease of visibility, explanatory images should be full width; expository or those with very simple subjects can be half-width, right aligned
+- For ease of visibility, images should be either full width, or 500 px if appropriate
 - Use PNG for all images, except photographs over 100-200 KB as a PNG (use moderately compressed JPEG); avoid sizes above 2000 x 2000 or below 1000 x 1000 if full width, or half of each if half width
+- GIFs should be compressed to 5 frames/s and should target a length of around 5-10 seconds, and be 500 px unlesss needed to illustrate a more complex or important subject
```
Comment (juanis2112) [inline STYLEGUIDE.md:77]: A lot of the gifs we already have are greater than 500 px. I would say most of them are 690 so this doesn't really make much sense.

File: STYLEGUIDE.md [hunk 2]
```diff
@@ -71,10 +72,11 @@ Must (not), should (not), recommended, suggested, may, and can generally follow
...
 - Always include alt text describing the content of the image
-- No space before, one space ("|") after if full width
+- No line break before, one line break ("|") after an image/GIF if it is at the end of a section
```
Comment (juanis2112) [inline STYLEGUIDE.md:79]: I think we have one blank line before and after every image and one line break ("|") followed by three blank lines if it is at the end of a section.

File: STYLEGUIDE.md (hunk 1 above)
Comment (CAM-Gerlach) [reply STYLEGUIDE.md:77]: Okay, sure. Thanks for pointing it out, I'll reword this.

File: STYLEGUIDE.md (hunk 2 above)
Comment (CAM-Gerlach) [reply STYLEGUIDE.md:79]: Yeah, this is extremely confusingly worded, my fault. I'll rewrite it.

File: STYLEGUIDE.md (hunk 1 above)
Comment (CAM-Gerlach) [reply STYLEGUIDE.md]: Reworded

File: STYLEGUIDE.md (hunk 2 above)
Comment (CAM-Gerlach) [reply STYLEGUIDE.md]: Reworded

**Comment by CAM-Gerlach** [comment]
I ended up doing a comprehensive rewrite on both the Style and Contributing guides, to ensure they were up to date and to address Juanita's feedback to hopefully make them more usable for new contributors.

File: STYLEGUIDE.md [hunk 3]
```diff
@@ -1,5 +1,9 @@
 # Spyder Documentation Style Guide
 
+This style guide is intended as a comprehensive reference on a wide variety of topics that may be pertinent to specific situations encountered when writing the docs.
+A more more digestible summary of the most important points for doc authors to actively keep in mind is included in the [Contributing Guide](https://github.com/spyder-ide/spyder-docs/blob/master/CONTRIBUTING.md).
```
Comment (juanis2112) [inline STYLEGUIDE.md:4]:
```suggestion
A more digestible summary of the most important points for doc authors to actively keep in mind is included in the [Contributing Guide](https://github.com/spyder-ide/spyder-docs/blob/master/CONTRIBUTING.md).
```
Delete repeated word

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -102,21 +74,37 @@ make.bat html
...
-* **UTF-8** for character encoding
-* **LF** for newlines
-* **ISO 8601** (YYYY-MM-DD HH:MM:SS) for dates/times
-* **SI/metric** for units
+* **Admonitions**: ``important::`` for key points, ``warnings`` for things to avoid, and ``note::`` for everything else.
+* **Blank lines**: One after all headings and before and after paragraphs, directives and ``|``s
+* **Filenames**: Lowercase, hyphen-separated
+* **Headings**: *Level 1*—Page titles, title case, ``###`` top and bottom; *Level 2*—3 blank lines before, Sentence case, ``===`` top and bottom; *Level 3*—2 blank lines before, Sentence case, ``~~~`` bottom; *Level 4*—1 blank line before, Sentence case, ``---`` bottom
+* **Images/GIFS**: Full width, 690 px or 500 px; manual break (``|``) after before a new section; PNGs or GIFs 5-10 s at 5 frames/s
+* **Indentation**: Spaces, no tabs; 3 spaces after directives, 4 in code blocks
```
Comment (juanis2112) [inline CONTRIBUTING.md:107]:
```suggestion
* **Indentation**: Spaces, no tabs; 3 spaces after directives, 4 inside code blocks
```

File: STYLEGUIDE.md (hunk 3 above)
Comment (CAM-Gerlach) [reply STYLEGUIDE.md:4]: Thanks, nice catch.

File: CONTRIBUTING.md (hunk 4 above)
Comment (CAM-Gerlach) [reply CONTRIBUTING.md:107]: Good call, thanks.

**Pull request merged**